lightning/src/sign/taproot.rs

```diff
@@ -0,0 +1,141 @@
+use bitcoin::blockdata::transaction::Transaction;
+use bitcoin::secp256k1;
+use bitcoin::secp256k1::{PublicKey, schnorr::Signature, Secp256k1, SecretKey};
+
+use musig2::types::{PartialSignature, PublicNonce};
+
+use crate::events::bump_transaction::HTLCDescriptor;
+use crate::ln::chan_utils::{ClosingTransaction, CommitmentTransaction, HolderCommitmentTransaction, HTLCOutputInCommitment};
+use crate::ln::PaymentPreimage;
+use crate::sign::ChannelSigner;
+
+pub trait TaprootSigner: ChannelSigner {
+	fn generate_local_nonce_pair(&self, secp_ctx: &Secp256k1<secp256k1::All>) -> PublicNonce;
+
+	/// Create a signature for a counterparty's commitment transaction and associated HTLC transactions.
+	///
+	/// Note that if signing fails or is rejected, the channel will be force-closed.
+	///
+	/// Policy checks should be implemented in this function, including checking the amount
+	/// sent to us and checking the HTLCs.
+	///
+	/// The preimages of outgoing HTLCs that were fulfilled since the last commitment are provided.
+	/// A validating signer should ensure that an HTLC output is removed only when the matching
+	/// preimage is provided, or when the value to holder is restored.
+	///
+	/// Note that all the relevant preimages will be provided, but there may also be additional
+	/// irrelevant or duplicate preimages.
+	//
+	// TODO: Document the things someone using this interface should enforce before signing.
+	fn partially_sign_counterparty_commitment(&self, commitment_tx: &CommitmentTransaction,
+		preimages: Vec<PaymentPreimage>, secp_ctx: &Secp256k1<secp256k1::All>,
+	) -> Result<(PartialSignature, Vec<Signature>), ()>;
```

Comment:
Do we expect this method to return `Signature` for each, if yes also the `inclusion_proof` for the script path as we expect this method to be use internally by LDK for on-chain claims?